src/scripts/main.js

```diff
@@ -1,3 +1,208 @@
+    case 'ArrowLeft': slideLeft(); break;
+    case 'ArrowRight': slideRight(); break;
+    case 'ArrowUp': slideUp(); break;
+    case 'ArrowDown': slideDown(); break;
+    default: break;
+  }
+
+  if (!isFieldChanged(field, fieldCopy)) {
+    setRundomNumber();
+  }
+
+  updateGame();
+});
+
+function updateGame() {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < cols; c++) {
+      const currentTile = table.rows[r].cells[c];
+      const valueOfTile = field[r][c];
+
+      currentTile.innerText = '';
+      currentTile.classList.value = '';
+      currentTile.className = `field-cell`;
+
+      if (valueOfTile > 0) {
+        currentTile.innerText = valueOfTile;
+        currentTile.classList.add(`field-cell--${valueOfTile}`);
+      }
+
+      if (valueOfTile === 2048) {
+        messageWin.classList.remove('hidden');
+        startButton.classList.replace('restart', 'start');
+        startButton.innerText = 'Start';
+      }
+    }
+  }
+  scoreGame.innerText = score;
+
+  if (isGameOver()) {
+    messageLose.classList.remove('hidden');
+    startButton.innerText = 'Start';
+  }
+};
+
+function filterZero(row) {
+  return row.filter(el => el !== 0);
+}
+
+function slide(row) {
+  let newRow = filterZero(row);
+
+  for (let i = 0; i < newRow.length - 1; i++) {
+    if (newRow[i] === newRow[i + 1]) {
+      newRow[i] *= 2;
+      newRow[i + 1] = 0;
+      score += newRow[i];
+    }
+  }
+  newRow = filterZero(newRow);
+
+  while (newRow.length < rows) {
+    newRow.push(0);
+  }
+
+  return newRow;
+};
+
+function slideLeft() {
+  for (let r = 0; r < rows; r++) {
+    let row = field[r];
+
+    row = slide(row);
+    field[r] = row;
+  }
+};
+
+function slideRight() {
+  for (let r = 0; r < rows; r++) {
+    let row = field[r].reverse();
+
+    row = slide(row);
+    field[r] = row.reverse();
+  }
+}
+
+function slideUp() {
+  for (let i = 0; i < cols; i++) {
+    let row = [field[0][i], field[1][i], field[2][i], field[3][i]];
+
+    row = slide(row);
+    field[0][i] = row[0];
+    field[1][i] = row[1];
+    field[2][i] = row[2];
+    field[3][i] = row[3];
+  }
+}
+
+function slideDown() {
+  for (let i = 0; i < cols; i++) {
+    let row = [field[0][i], field[1][i], field[2][i], field[3][i]].reverse();
+
+    row = slide(row).reverse();
+    field[0][i] = row[0];
+    field[1][i] = row[1];
+    field[2][i] = row[2];
+    field[3][i] = row[3];
+  }
+}
+
+function isGameOver() {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < cols; c++) {
+      if (field[r][c] === 0) {
+        return false;
+      }
+    }
+  }
+
+  for (let r = 0; r < rows - 1; r++) {
+    for (let c = 0; c < cols; c++) {
+      if (field[r][c] === field[r + 1][c]) {
+        return false;
+      }
+    }
+  }
+
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < cols - 1; c++) {
+      if (field[r][c] === field[r][c + 1]) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+}
+
+function isFieldChanged(fieldGame, fieldCopy) {
```

Comment:
Function name 'isFieldChanged' is confusing. Consider renaming it to 'isFieldUnchanged' since it returns 'false' if the fields are the same.